server/enhancedDate.js

```diff
@@ -1 +1,59 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+
+var dateIsSet;
+var dayOfWeekArray = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+var monthArray = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+
+//default function
+function isDateSet() {
+	if(!dateIsSet) {console.warn('date hasn\'t been set! Please call setDate first!'); return false;}
+}
+
+function today(input) {
+	var day = input.getDay();
+	var month = input.getMonth();
+	var year = input.getFullYear();
+	return "Date : " + day + month + year;
+}
+
+//returned function
+function setDate(date) {
```

Comment:
You have all the right logic checks here - my only feedback on this function would be to find a way to structure it so you have a single return statement. The code is technically sound, but when multiple points where a function can return reduces the readability of the code.